core/src/main/java/org/infinispan/interceptors/compat/TypeConverterStream.java

```diff
@@ -0,0 +1,66 @@
+package org.infinispan.interceptors.compat;
+
+import org.infinispan.CacheStream;
+import org.infinispan.commons.util.IteratorMapper;
+import org.infinispan.compat.TypeConverter;
+import org.infinispan.container.InternalEntryFactory;
+import org.infinispan.container.entries.CacheEntry;
+import org.infinispan.stream.impl.AbstractDelegatingCacheStream;
+import org.infinispan.stream.impl.spliterators.IteratorAsSpliterator;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+
+/**
+ * Delegating stream that converts variables.  Note this stream specifically doesn't specify generics as it can have
```

Comment:
Yeah I was trying not to say values since I thought it might have caused confusion with the CacheEntry at the time since it does both key and values.  But I can change it.